packages/core/src/i18n/i18n-setup.js

```diff
@@ -0,0 +1,39 @@
+import i18next from 'i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
+
+import mainTag from './language-detector';
+import enTranslation from './translations/en';
+import esTranslation from './translations/es';
+
+const languageDetector = new LanguageDetector();
+languageDetector.addDetector(mainTag);
+i18next.use(languageDetector).init({
+  fallbackLng: 'en',
+  detection: {
+    order: ['mainTag', 'htmlTag'],
+  },
+  resources: {
+    en: { translation: enTranslation },
+    es: { translation: esTranslation },
+  },
+});
+
+export { i18next };
+
+window.addEventListener('DOMContentLoaded', event => {
+  console.log('DOM fully loaded and parsed');
+
+  const element = document.querySelector('main');
+
+  const observer = new MutationObserver(mutations => {
+    mutations.forEach(mutation => {
+      if (mutation.type === 'attributes' && mutation.attributeName === 'lang') {
+        i18next.changeLanguage(element.getAttribute('lang'));
+      }
+    });
+  });
+
+  observer.observe(element, {
+    attributes: true,
+  });
+});
```

Comment:
This sets up a MutationObserver to change the language in the `i18next` dependency anytime the lang in `<main lang="">` changes. This is what supports the dynamic translation.